From e3c6daa068b6fc8c7ad01c71498f136c7a9aa001 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Wed, 4 May 2011 13:08:58 +1000
Patch-mainline: 3.0-rc3
References: fate#310916
Git-commit: cb3c438ea4c0e638cba08bfaa8e2b64ffb8a6ce3
Subject: [PATCH 2384/2588] drm/fb: fix CONFIG_DRM=m && CONFIG_FB=n

The previous commit to move the parsing into the core drm created a
new situation and a soft dependency on the CONFIG_FB. We really don't
want to make this a hard dependency so just wrap the one place that
actually needs an fb symbol.

Reported-by: Randy Dunlap <randy.dunlap@oracle.com>
Signed-off-by: Dave Airlie <airlied@redhat.com>
(cherry picked from commit cb3c438ea4c0e638cba08bfaa8e2b64ffb8a6ce3)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/drm_modes.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/drm_modes.c b/drivers/gpu/drm/drm_modes.c
index 207b7eb..c2d32f2 100644
--- a/drivers/gpu/drm/drm_modes.c
+++ b/drivers/gpu/drm/drm_modes.c
@@ -1000,8 +1000,10 @@ bool drm_mode_parse_command_line_for_connector(const char *mode_option,
 	int i;
 	enum drm_connector_force force = DRM_FORCE_UNSPECIFIED;
 
+#ifdef CONFIG_FB
 	if (!mode_option)
 		mode_option = fb_mode_option;
+#endif
 
 	if (!mode_option) {
 		mode->specified = false;
-- 
1.7.6

